src/openforms/authentication/utils.py

```diff
@@ -17,15 +18,10 @@ class FormAuth(TypedDict):
 
 
 def store_auth_details(submission: Submission, form_auth: FormAuth) -> None:
-    plugin = form_auth["plugin"]
     attribute = form_auth["attribute"]
 
     assert (
         attribute in AuthAttribute.values
     ), f"Unexpected auth attribute {attribute} specified"
 
-    # DO NOT log this in plain text, it is considered sensitive information (BSN for example)
-    identifier = form_auth["value"]
-    submission.auth_plugin = plugin
-    setattr(submission, attribute, identifier)
-    submission.save(update_fields=["auth_plugin", attribute])
+    AuthInfo.objects.update_or_create(submission=submission, **form_auth)
```

Comment:
```suggestion
    AuthInfo.objects.update_or_create(submission=submission, defaults=form_auth)
```
so that it only matches on `submission` and not the combination of all kwargs :)